@here/harp-materials/lib/SkyAtmosphereMaterial.ts

```diff
@@ -0,0 +1,423 @@
+
+/**
+ * `SkyAtmosphereShader`.
+ *
+ * Describes shading of atmosphere as seen from outer space.
+ */
+export const SkyAtmosphereShader: THREE.Shader = {
+    uniforms: {
+        u_eyePositionWorld: new THREE.Uniform(new THREE.Vector3()),
+        u_lightDirectionWorld: new THREE.Uniform(new THREE.Vector3(0, 1, 0)),
+        u_modelViewProjection: new THREE.Uniform(new THREE.Matrix4()),
+        // Environment settings:
+        // atmosphere inner and outer radius, camera height, light mode
+        u_atmosphereEnv: new THREE.Uniform(
+            new THREE.Vector4(
+                // Maximum inner radius
+                EQUATORIAL_RADIUS,
+                // Maximum outer radius
+                EQUATORIAL_RADIUS * 1.025,
+                // Camera height
+                0,
+                // Toggles the light modes:
+                // 0 - light always directly overhead,
+                // 1 - lighting uses light direction: uniform u_lightDirectionWorld
+                1
+            )
+        ),
+        u_hsvCorrection: new THREE.Uniform(new THREE.Vector3(0, 0, 0)),
+
+        topColor: new THREE.Uniform(new THREE.Color(0x0077ff)),
+        bottomColor: new THREE.Uniform(new THREE.Color(0xffffff)),
+        offset: new THREE.Uniform(33.0),
+        exponent: new THREE.Uniform(0.6),
+        fogColor: new THREE.Uniform(new THREE.Color(0x0077ff)), // scene.fog.color
+        fogNear: new THREE.Uniform(new THREE.Color(0x0077ff)), // scene.fog.near
+        fogFar: new THREE.Uniform(new THREE.Color(0xffffff)) // scene.fog.far
+    },
+
+    vertexShader: `
+    precision highp float;
+    precision highp int;
+
+    attribute vec4 position;
+
+    uniform mat4 u_modelViewProjection;
+    uniform vec3 u_eyePositionWorld;
+    uniform vec3 u_lightDirectionWorld;
+
+    uniform vec4 u_atmosphereEnv; // Atmosphere inner and outer radius, camera height, light mode
+    uniform vec3 u_hsvCorrection;
+
+    const float Pi = 3.141592653589793;
+    const float Kr = 0.0025;
+    const float Kr4PI = Kr * 4.0 * Pi;
+    const float Km = 0.0015;
+    const float Km4PI = Km * 4.0 * Pi;
+    const float ESun = 15.0; // should be 15.0
+    const float KmESun = Km * ESun;
+    const float KrESun = Kr * ESun;
+    const vec3 InvWavelength = vec3(
+        5.60204474633241,  // Red = 1.0 / Math.pow(0.650, 4.0)
+        9.473284437923038, // Green = 1.0 / Math.pow(0.570, 4.0)
+        19.643802610477206); // Blue = 1.0 / Math.pow(0.475, 4.0)
+
+    const int nSamples = 2;
+    const float fSamples = 2.0;
+
+    varying vec3 v_rayleighColor;
+    varying vec3 v_mieColor;
+    varying vec3 v_vertToCamera;
+
+    ${AtmosphereShaderChunks.atmosphere_common_utils}
+    ${AtmosphereShaderChunks.atmosphere_vertex_utils}
+
+    //
+    // Computes rayleight and mia atmosphere factors for sky.
+    //
+    // Code based on GPU Gems article.
+    //
+    // Author: Sean O'Neil
+    //
+    // Copyright (c) 2004 Sean O'Neil
```

Comment:
What kind of license does the shader from GPU gems use? I think we need to check if it is compatible with our Apache 2 license.

Reply:
Yeah, already reported that to Stephan, so he will take care of that.